mailpoet/lib/WooCommerce/Integrations/AutomateWooHooks.php

```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\WooCommerce\Integrations;
+
+use MailPoet\Entities\SubscriberEntity;
+use MailPoet\Subscribers\SubscribersRepository;
+use MailPoet\WP\Functions as WPFunctions;
+
+class AutomateWooHooks {
+  const AUTOMATE_WOO_PLUGIN_SLUG = 'automatewoo/automatewoo.php';
+
+  /** @var SubscribersRepository */
+  private $subscribersRepository;
+
+  /** @var WPFunctions */
+  private $wp;
+
+  public function __construct(
+    SubscribersRepository $subscribersRepository,
+    WPFunctions $wp
+  ) {
+    $this->subscribersRepository = $subscribersRepository;
+    $this->wp = $wp;
+  }
+
+  public function isAutomateWooActive(): bool {
+    return $this->wp->isPluginActive(self::AUTOMATE_WOO_PLUGIN_SLUG);
+  }
+
+  public function areMethodsAvailable(): bool {
+    return class_exists('AutomateWoo\Customer_Factory') && method_exists('AutomateWoo\Customer_Factory', 'get_by_email') &&
+      class_exists('AutomateWoo\Customer') && method_exists('AutomateWoo\Customer', 'opt_out');
+  }
+
+  /**
+   * @return \AutomateWoo\Customer|false
+   */
+  public function getAutomateWooCustomer(string $email) {
+    // AutomateWoo\Customer_Factory::get_by_email() returns false if customer is not found
+    // Second parameter is set to false to prevent creating new customer if not found
+    return \AutomateWoo\Customer_Factory::get_by_email($email, false);
+  }
+
+  public function setup(): void {
+    if (!$this->isAutomateWooActive() || !$this->areMethodsAvailable()) {
+      return;
+    }
+    $this->wp->addAction(SubscriberEntity::HOOK_SUBSCRIBER_STATUS_CHANGED, [$this, 'maybeOptOutSubscriber'], 10, 1);
+    $this->wp->addAction('mailpoet_woocommerce_segment_unsubscribed', [$this, 'optOutSubscriber'], 10, 1);
+  }
+
+  public function optOutSubscriber($subscriber): void {
+    if (!$this->isAutomateWooActive() || !$this->areMethodsAvailable()) {
+      return;
+    }
+
+    $automateWooCustomer = $this->getAutomateWooCustomer($subscriber->getEmail());
+    if (!$automateWooCustomer) {
+      return;
+    }
+
+    $automateWooCustomer->opt_out();
+  }
+
+  public function maybeOptOutSubscriber(int $subscriberId): void {
+    $subscriber = $this->subscribersRepository->findOneById($subscriberId);
+    if (!$subscriber || !$subscriber->getEmail() || $subscriber->getStatus() !== SubscriberEntity::STATUS_UNSUBSCRIBED) {
```

Comment:
I remember we chatted about supporting opting out of Automate Woo when the subscriber is deleted, and we were wondering what happens when a subscriber is deleted (if it is removed from the database or if we just set the deleted_at field). Since this use case was not mentioned in the PR description, I'm assuming we intentionally decided not to support it for now. I'm mentioning it here just to confirm that it was a conscious decision.

Reply:
Thanks for asking.
Yes, it was intentional. The PR only handles the changes in global status and the checkbox on checkout.
When a subscriber is deleted, we only set the deleted_at field. The global status does not change.

Reply:
:+1: